src/main/java/edu/wpi/grip/core/Connection.java

```diff
@@ -79,4 +104,13 @@ public void removeConnection(SourceRemovedEvent e) {
             }
         }
     }
+
+    /**
+     * Thrown when the user tries to create a connection that will result in an infinite loop.
+     */
+    public static class InfiniteLoopException extends Error {
```

Comment:
I'm not sure this should be an exception.  Exceptions are for signaling error conditions, not input validation.

A better solution might be to use the `willCreateInfiniteLoop` method to prevent users from creating the loops in the first place instead of printing an exception if they do.